src/huggingface_hub/hf_api.py

```diff
@@ -7906,6 +7909,12 @@ def update_inference_endpoint(
                 Inference Endpoint running on the `text-generation-inference` (TGI) framework (see examples).
             secrets (`Dict[str, str]`, *optional*):
                 Secret values to inject in the container environment.
+
+            domain (`str`, *optional*):
+                The custom domain for the Inference Endpoint (e.g. `"api-inference.endpoints.huggingface.tech"`).
```

Comment:
The custom domain must belong to the user and have been created using this api call beforehand:
https://api.endpoints.huggingface.cloud/#post-/v2/network/domain/-namespace-

Reply:
it can either be:
- one we reserve on the fly in our dns zone custom-whatever.us-east-1.aws.endpoints.huggingface.cloud (if not already taken) 
- or one the customer brings to us (then they should cname it to custom-domains.endpoints.huggingface.cloud)

Reply:
since it's optional, maybe we don't need to elaborate much? wdyt?

Reply:
yeah sure, need to document better the feature anyway but you're right this should not be done here in the docstring